Update supported platforms

jsf9k opened the pull request:
## 🗣 Description ##

This pull request updates the platforms supported by this Ansible role in addition to ensuring that the [cyhy-feeds](https://github.com/cisagov/cyhy-feeds) package is installed with `pip2`.

## ⚠ Note ##

This pull request is reliant on the merge of the following PRs:

- cisagov/ansible-role-cyhy-feeds#8
- cisagov/ansible-role-pip#50
- cisagov/ansible-role-python#46

## 💭 Motivation and context ##

We are in the process of updating the instances in [cisagov/cyhy_amis](https://github.com/cisagov/cyhy_amis) to use Debian Buster for instances running Python 2 code. With the above mentioned PRs for Python/pip installation we can expand the supported platforms.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert dependencies to default branches.

jsf9k committed:
Add support for more platforms

We can only support some platforms, since Python 2 is EOL and support
is being removed from Linux distributions.

jsf9k committed:
Ensure that cisagov/cyhy-feeds is installed via Python 2

Also ensure that this Ansible role works with the forthcoming code in
cisagov/ansible-role-python#46 and cisagov/ansible-role-pip#50.

dav3r reviewed (approved):
Beautiful 🌷

mcdonnnj reviewed (commented):
So an important thing I wanted to mention here is that [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) is in a [partial Python 2 requirement state](https://github.com/cisagov/cyhy-feeds#getting-started). We are running a specific piece on Python 2 because that piece has not been updated to Python 3, but the [cisagov/mongo-db-from-config](https://github.com/cisagov/mongo-db-from-config) package it uses [only explicitly supports Python 3](https://github.com/cisagov/mongo-db-from-config/blob/021bb2ceae01b918f863592269899185f17ac2f4/setup.py#L35-L50). Do we want to look at cleaning up [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) to clearly support either both Python 2 and Python 3 or *just* Python 3?

jsf9k commented:
> So an important thing I wanted to mention here is that [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) is in a [partial Python 2 requirement state](https://github.com/cisagov/cyhy-feeds#getting-started). We are running a specific piece on Python 2 because that piece has not been updated to Python 3, but the [cisagov/mongo-db-from-config](https://github.com/cisagov/mongo-db-from-config) package it uses [only explicitly supports Python 3](https://github.com/cisagov/mongo-db-from-config/blob/021bb2ceae01b918f863592269899185f17ac2f4/setup.py#L35-L50). Do we want to look at cleaning up [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) to clearly support either both Python 2 and Python 3 or _just_ Python 3?

If this code is working as it is then I think we should create an issue and move on.  If that isn't palatable, then making this repo all Python 3 is probably the way to go.  I'd like that to be done in a separate PR, though.

mcdonnnj commented:
> > So an important thing I wanted to mention here is that [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) is in a [partial Python 2 requirement state](https://github.com/cisagov/cyhy-feeds#getting-started). We are running a specific piece on Python 2 because that piece has not been updated to Python 3, but the [cisagov/mongo-db-from-config](https://github.com/cisagov/mongo-db-from-config) package it uses [only explicitly supports Python 3](https://github.com/cisagov/mongo-db-from-config/blob/021bb2ceae01b918f863592269899185f17ac2f4/setup.py#L35-L50). Do we want to look at cleaning up [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) to clearly support either both Python 2 and Python 3 or _just_ Python 3?
> 
> If this code is working as it is then I think we should create an issue and move on. If that isn't palatable, then making this repo all Python 3 is probably the way to go. I'd like that to be done in a separate PR, though.

I'm alright with making an issue and circling back later, but I would like to remove the language about *requiring* Python 2 and make it clear it's a preference. I would need to test to confirm, but I'm pretty sure at this point the library is riding Py2/Py3 dual compatibility (at least syntactically).

jsf9k commented:
> So an important thing I wanted to mention here is that [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) is in a [partial Python 2 requirement state](https://github.com/cisagov/cyhy-feeds#getting-started). We are running a specific piece on Python 2 because that piece has not been updated to Python 3, but the [cisagov/mongo-db-from-config](https://github.com/cisagov/mongo-db-from-config) package it uses [only explicitly supports Python 3](https://github.com/cisagov/mongo-db-from-config/blob/021bb2ceae01b918f863592269899185f17ac2f4/setup.py#L35-L50). Do we want to look at cleaning up [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) to clearly support either both Python 2 and Python 3 or _just_ Python 3?

See cisagov/cyhy-feeds#43, which is an attempt to make [cisagov/cyhy-feeds](https://github.com/cisagov/cyhy-feeds) fully support Python 3 (and only Python 3).

mcdonnnj committed:
Revert branches used for testing

These branches have been merged into their respective develop branches.

mcdonnnj committed:
Pin the version of cisagov/mongo-db-from-config installed

The pin is the last commit before the project became Python 3 only.

mcdonnnj committed:
Adjust molecule test

Adjust the test that checks installed Python packages to only check the
packages we are explicitly installing.

mcdonnnj commented:
@jsf9k Would you look the changes over and see if you have any contention? This is just to get this PR through on Python 2 until we're ready to merge the instance this runs on bumping to Debian Buster (which will allow changing this project to Python 3 only).

dav3r reviewed (approved):
This all looks good to my eyes 👍

jsf9k reviewed (commented):
LGTM

mcdonnnj reviewed (approved):
LGTM
![image](https://user-images.githubusercontent.com/50747025/193109703-aba5681b-9260-4139-bd37-a815bf1e85fd.png)
